controllers/withwatcher/watcher_controller_helper_test.go

```diff
@@ -80,54 +85,58 @@ func createWatcherCR(managerInstanceName string, statusOnly bool) *v1alpha1.Watc
 				fmt.Sprintf("%s-watchable", managerInstanceName): "true",
 			},
 			Field: field,
+			Gateway: &v1alpha1.GatewayConfig{
+				LabelSelector: v1alpha1.DefaultIstioGatewaySelector(),
+			},
 		},
 	}
 }
 
-func listTestWatcherCrs(kcpClient client.Client) []*v1alpha1.Watcher {
+func listTestWatcherCrs() []*v1alpha1.Watcher {
+	watcherList := &v1alpha1.WatcherList{}
 	watchers := make([]*v1alpha1.Watcher, 0)
-	for _, component := range centralComponents {
-		watcherCR := &v1alpha1.Watcher{}
-		err := kcpClient.Get(suiteCtx, client.ObjectKey{
-			Name:      fmt.Sprintf("%s-sample", component),
-			Namespace: metav1.NamespaceDefault,
-		}, watcherCR)
-		if err != nil {
-			continue
-		}
-
-		watchers = append(watchers, watcherCR)
+	if err := controlPlaneClient.List(suiteCtx, watcherList); err != nil {
+		return watchers
+	}
+	for i := range watcherList.Items {
+		watchers = append(watchers, &watcherList.Items[i])
 	}
 	return watchers
 }
 
-func isCrDeletionFinished(watcherObjKeys ...client.ObjectKey) func(g Gomega) bool {
-	if len(watcherObjKeys) > 1 {
+func isCrDeleted(watcherObjKeys client.ObjectKey) error {
+	err := controlPlaneClient.Get(suiteCtx, watcherObjKeys, &v1alpha1.Watcher{})
+	if apierrors.IsNotFound(err) {
 		return nil
 	}
-	if len(watcherObjKeys) == 0 {
-		return func(g Gomega) bool {
-			watchers := listTestWatcherCrs(controlPlaneClient)
-			return len(watchers) == 0
-		}
-	}
-	return func(g Gomega) bool {
-		err := controlPlaneClient.Get(suiteCtx, watcherObjKeys[0], &v1alpha1.Watcher{})
-		return apierrors.IsNotFound(err)
+	return err
```

Comment:
if the cr is present then the get call will not error out. This means that it will return nil even if it is still there

Reply:
Yes, I changed the condition to check `if !apierrors.IsNotFound(err)`, to make sure only the not found case can continue.